Assets/Scripts/Drone Scripts/Drone.cs

```diff
@@ -242,14 +247,37 @@ public void DeleteAllWaypoints()
             }
         }
         
+
+
 		/******************************/
-		//  WAYPOINTS GETTER METHODS  //
+		//       HELPER METHODS       //
 		/******************************/
 
         /// <summary>
-        /// Return the length of the waypoints list
+        /// Return true if waypoint list is empty and false if not
         /// </summary>
+        /// <param name="waypointList"></param>
         /// <returns></returns>
+        private bool isEmptyWaypointList(List<Waypoint> waypointList)
+        {
+            if (waypointList == null)
+            {
+                return true;
+            }
+            return waypointList.Count == 0;
```

Comment:
When would there be a case where the list is empty but it's count is more than 0?

Reply:
there might be a case initially where the list variable is null, therefore it's done this way.
@apollovision any other reason?